service/services/services.go

```diff
@@ -0,0 +1,664 @@
+		return err
+	}
+	tableName = tableConf.ActualTable
+	e, err := utils.CreateConditionExpression(condExpression, attrMap)
+	if err != nil {
+		return err
+	}
+	return storage.GetStorageInstance().SpannerDelete(ctx, tableName, primaryKeyMap, e, expr)
+}
+
+//BatchDelete service
+func BatchDelete(ctx context.Context, tableName string, keyMapArray []map[string]interface{}) error {
+	tableConf, err := config.GetTableConf(tableName)
+	if err != nil {
+		return err
+	}
+	oldRes, _ := BatchGet(ctx, tableName, keyMapArray)
+
+	tableName = tableConf.ActualTable
+	err = storage.GetStorageInstance().SpannerBatchDelete(ctx, tableName, keyMapArray)
+	if err != nil {
+		return err
+	}
+	go func() {
+		if len(oldRes) == len(keyMapArray) {
+			for i := 0; i < len(keyMapArray); i++ {
+				go StreamDataToThirdParty(oldRes[i], keyMapArray[i], tableName)
+			}
+		} else {
+			for i := 0; i < len(keyMapArray); i++ {
+				go StreamDataToThirdParty(nil, keyMapArray[i], tableName)
+			}
+
+		}
+	}()
+	return nil
+}
+
+//Scan service
+func Scan(ctx context.Context, scanData models.ScanMeta) (map[string]interface{}, error) {
+	query := models.Query{}
+	query.TableName = scanData.TableName
+	query.Limit = scanData.Limit
+	if query.Limit == 0 {
+		query.Limit = 5000
+	}
+	query.StartFrom = scanData.StartFrom
+	query.RangeValMap = scanData.ExpressionAttributeMap
+	query.IndexName = scanData.IndexName
+	query.FilterExp = scanData.FilterExpression
+	query.ExpressionAttributeNames = scanData.ExpressionAttributeNames
+	query.OnlyCount = scanData.OnlyCount
+	query.ProjectionExpression = scanData.ProjectionExpression
+
+	for k, v := range query.ExpressionAttributeNames {
+		query.FilterExp = strings.ReplaceAll(query.FilterExp, k, v)
+	}
+
+	rs, _, err := QueryAttributes(ctx, query)
+	return rs, err
+}
+
+func scanSpanerTable(ctx context.Context, tableName, pKey, sKey string) ([]map[string]interface{}, error) {
+
+	var startFrom map[string]interface{}
+	var result []map[string]interface{}
+	query := models.Query{}
+	query.TableName = tableName
+	var originalLimit int64 = 5000
+	query.Limit = originalLimit + 1
+	for {
+		query.StartFrom = startFrom
+		stmt, cols, isCountQuery, offset, _, err := createSpannerQuery(&query, pKey, pKey, sKey)
+		if err != nil {
+			return nil, err
+		}
+		resp, err := storage.GetStorageInstance().ExecuteSpannerQuery(ctx, query.TableName, cols, isCountQuery, stmt)
+		if err != nil {
+			return nil, err
+		}
+		lastIndex := len(resp) - 1
+		pVal, ok := resp[lastIndex][pKey]
+		if !ok {
+			return nil, errors.New("ResourceNotFoundException")
+		}
+		startFrom = map[string]interface{}{pKey: pVal, "offset": originalLimit + offset}
+		result = append(result, resp...)
+		if len(resp) < int(originalLimit) {
+			break
+		}
+	}
+	return result, nil
+}
+
+func Remove(ctx context.Context, tableName string, updateAttr models.UpdateAttr, actionValue string, expr *models.UpdateExpressionCondition, oldRes map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Forget me if I am asking a simple question: what is the difference between `Delete`, `Remove`, `Del`? Can we have more docs to describe those operations?